### DISCUSSION TITLE
Problems with the axes Configuration XZBC

### DISCUSSION THREAD

**Discussion by GrblGru** [fra589/grbl-Mega-5X#356] [Q&A] [answered]
Hello everyone,

I am having problems creating the axis configuration XZBC  for a lathe. (Mega-5X Version = 1.2.h) 
Since I use the returned axis values to display my graphics, I would like the values to be delivered in the order XZBC.
The control of the X, Z and B axes works and the returned values are also sent in the correct order. However, Mega-5X does not recognize the C-axis. (Error 20).  Instead, Y commands are recognized and also sent back in the fourth position.

Hints for solving the problem are welcome :)

Thanks
GrblGru

I use following config.h

#define N_AXIS 4        // Number of axes (3 to 6)
#define N_AXIS_LINEAR 2 // Number of linears axis, must be <= N_AXIS

// Axis indexing and names
#define AXIS_1 0        // Axis indexing value. Must start with 0 and be continuous.
#define AXIS_1_NAME 'X' // Axis names must be in X, Y, Z, A, B, C, U, V, W, D, E & H.
#define AXIS_2 1
#define AXIS_2_NAME 'Z'
#define AXIS_3 2
#define AXIS_3_NAME 'Y'
#if N_AXIS <3
  #error "N_AXIS must be >= 3. N_AXIS < 3 is not implemented."
#endif
#if N_AXIS > 3
  #define AXIS_4 3
  #define AXIS_4_NAME 'B' // Letter of axis number 4
#endif
#if N_AXIS > 4
  #define AXIS_5 4
  #define AXIS_5_NAME 'C' // Letter of axis number 5
#endif
#if N_AXIS > 5
  #define AXIS_6 5
  #define AXIS_6_NAME 'A' // Letter of axis number 6
#endif
#if N_AXIS > 6
  #error "N_AXIS must be <= 6. N_AXIS > 6 is not implemented."
#endif

// Renaming axis doesn't change their number. By default, the status report give axis values in
// the order of their number. Some graphical interface are not able to affect axis values reported
// by Grbl to the correct axis name.
// Uncomment to enable sorting of axis values by axis_names rather than by axis number. Default disabled.
// If this option is enabled, the sorting order will be X, Y, Z, U, V, W, A, B, C, D, E & H 
// as defined below.
#define SORT_REPORT_BY_AXIS_NAME
#define AXIS_NAME_SORT_ORDER {'X', 'Z', 'B', 'C', 'A', 'Y'}

**Comment by fra589** [comment, marked as answer]
Hi @GrblGru,  
  
In the config.h extract that you put, I still see the definition `#define AXIS_3_NAME 'Y'`.
Then, with `#define N_AXIS 4`, the elements: `#define AXIS_5 4` and `#define AXIS_5_NAME 'C'` are not taken into account.

Finaly, for your need, you must have:
```
#define AXIS_1_NAME 'X'
(...)
#define AXIS_2_NAME 'Z'
(...)
#define AXIS_3_NAME 'B'
(...)
#define AXIS_4_NAME 'C'
(...)
#define AXIS_NAME_SORT_ORDER {'X', 'Z', 'B', 'C'}
```

@++;
Gauthier.

**Reply by GrblGru** [reply]
That works :)
You are the best, thank you my friend.

GrblGru
